### PR TITLE
Add  --init-blockchain and update to run on nightly cluster

### PR DESCRIPTION
**Pull request by sbtaylor15**
## Description

This PR uses the latest image with --listen-only --init-blockchain parameters.  The logic in the entrypoint.sh and helm charts have been modified to handle running on the nightly cluster (outside of boot.pyrsia.link).   Looping was added to wait for resources to be available and valid before moving onto the next step in bringing up the nodes.

Fixes pyrsia/pyrsia#904
Fixes pyrsia/pyrsia#1114

## PR Checklist

- [X] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [X] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/get_involved/good_pr.md)
- [X] I've included a good title and brief description along with how to review them.
- [X] I've linked any associated an [issue](https://github.com/pyrsia/pyrsia/issues).

### Code Contributions

- [X] I've built the code `cargo build --all-targets` successfully.
- [X] I've run the unit tests `cargo test --workspace` and everything passes.
- [X] I've made sure my rust toolchain is current `rustup update`.

### CODE REVIEW

**Review by erwin1 — APPROVED** [review]

**Review by fishseabowl — APPROVED** [review]
lgtm

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -146,7 +146,7 @@ jobs:
       if: github.repository_owner == 'pyrsia' && (github.event_name == 'push' || github.event_name == 'release')
       run: |
         sudo go install github.com/mikefarah/yq/v4@latest
-        yq e '.image.tag |= "0.1.0+${{ github.run_number }}"' -i installers/helm/pyrsia-node/values.yaml
+        yq e '.image.tag |= "0.1.0-${{ github.run_number }}"' -i installers/helm/pyrsia-node/values.yaml
```
Comment (betarelease) [inline .github/workflows/rust.yml:149]: was the - intended?

File: installers/docker/node-entrypoint.sh [hunk 2]
```diff
@@ -10,17 +11,48 @@ if [ -f /var/run/secrets/kubernetes.io/serviceaccount/namespace ]; then
...
-        # I am not pyrsia-node-0.pyrsia.link so use boot.pyrsia.link for boot address and connect to it
-        /usr/bin/pyrsia_node $*
+        exit $?
     fi
-else
-    # not running under k8s so just startup in default mode
-    /usr/bin/pyrsia_node $*
 fi
+
+# I am not node-0 so use boot.pyrsia.link for boot address and connect to it
```
Comment (betarelease) [inline installers/docker/node-entrypoint.sh:46]: A lot of this should move into rust code sometime later. But it is ok for now I think.

File: installers/docker/node-entrypoint.sh [hunk 3]
```diff
@@ -10,17 +11,48 @@ if [ -f /var/run/secrets/kubernetes.io/serviceaccount/namespace ]; then
     CACERT=${SERVICEACCOUNT}/ca.crt
     PODNAME=`hostname`
     export PYRSIA_EXTERNAL_IP=$(curl -s --cacert ${CACERT} --header "Authorization: Bearer ${TOKEN}" -X GET ${APISERVER}/api/v1/namespaces/${NAMESPACE}/services/${PODNAME} | jq -r ".status.loadBalancer.ingress[0].ip")
+
+    # Wait for the service to be assigned an ip addr
```
Comment (betarelease) [inline installers/docker/node-entrypoint.sh:15]: We should move all this logic to the rust code as well sometime. It is starting to get complex here.

**Review by betarelease — APPROVED** [review]
Please verify that the - in the version numbering was intended.

File: .github/workflows/rust.yml (hunk 1 above)
Comment (sbtaylor15) [reply .github/workflows/rust.yml:149]: Yes, you need a dash for docker tags.  docker doesn't support +

File: installers/docker/node-entrypoint.sh (hunk 2 above)
Comment (sbtaylor15) [reply installers/docker/node-entrypoint.sh:46]: @betarelease see issue #1110

**Pull request merged**